RELEASE.md

```diff
@@ -72,8 +72,18 @@ To release new versions of ThriftRW Go, follow these instructions.
 
 9.  Land the pull request after approval as a **merge commit**. To do this,
     select **Create a merge commit** from the pull-down next to the merge
-    button and click **Merge pull request**. Make sure you delete that branch
-    after it has been merged with **Delete Branch**.
+    button and click **Merge pull request**.
+
+    Please update the subject and body of the merge commit with a short
+    description of the changes to be merged into master.
```

Comment:
We can state that the subject should be `Release v1.2.3 (#PR)` (like the back to dev merge).

The body is more free-form.